x-team/auto-ui
# Adjust AdminScreen and Modal for TwoColumnsLayout integration

**Release Type:** *non-breaking feature* 

Fixes https://x-team-internal.atlassian.net/browse/XP-2833

## Description

This PR have adjustments on `AdminScreen` and `Modal` in order to make it compatible with requirements of `TwoColumnsLayout`.

## Checklist


## Related PRs

branch | PR
------ | ------
`XP-2833-adjust-adminscreen-and-modal-for-twocolumns-layout-integration` | [Auto](https://github.com/x-team/auto/pull/1822)

## Steps to Test or Reproduce

1. Go to `AdminScreen` stories
   1. Observe that the header is always fixed at the top of the page
   1. Observe that when the content heigh is too tall, it's correct scrollable
   1. Observe that when the content heigh is short it does't display scrollbars in the page
1. Go to `TwoColumnsLayout` stories composed with `AdminScreen`
   1. Observe that it doesn't overflow the page viewport anymore
   1. Observe that the layout occupy full available height space of the screen
   1. Observe that sidebar and content have independent vertical and horizontal scrolling when needed
1. Go to `ListsScreen` stories composed with `AdminScreen`
  1. Observe the this PR doesn't introduced any regression
1. Go to `Modal` stories and observe that is working as expected
   1. Clicking on overlay or in the `X` should close
   1. Observe that the `Modal` showcases also scrolls well and have it's content visible when scrolling
   1. Observe that when the `Modal` is opened the page content is not scrollable
   1. Observe that when closes the `Modal` the page content is scrollable again
   1. On `AdminScreen`, `ListsScreen` and `TwoColumnsLayout` showcases have demonstrations with `Modal` open, observe that the `Modal` behaviour is correct there

## Impacted Areas in Application

- AdminScreen
- Modal

## Screenshots

_A GIF that fully demonstrates the fix is too big to be attached here, so it will be posted in the Jira ticket. The fixes can be viewed in the stories:_

<img width="378" alt="screen shot 2019-02-18 at 10 47 06" src="https://user-images.githubusercontent.com/131859/52955133-b489f200-336a-11e9-8880-a1724ba28e7b.png">
